lib/seattleflu/id3c/cli/command/etl/redcap_det_fh_airs.py

```diff
@@ -0,0 +1,910 @@
+                    f"{redcap_record_instance.get('subject_id')}, "
+                    "encounter: "
+                    f"{redcap_record_instance.get('event_name')}: "
+                    "insufficient information to construct encounter")
+                continue
+        else:
+            LOG.info(f"Skipping event: {redcap_record_instance.event_name!r} for record "
+            f"{redcap_record_instance.get('subject_id')} because the event is not one "
+            "that we process")
+            continue
+
+        # Skip an ENCOUNTER instance if we don't have the data we need to
+        # create an encounter.
+        # In this study they may send up to 2 AIRS kits per weekly encounter,
+        #  so I guess we need to check both
+        # TODO: verify with study that this is irrelevant if we only check if 'weekly' is complete, above.
+        if event_type == EventType.ENCOUNTER:
+            if not is_complete('weekly', redcap_record_instance) \
+                or not collection_method:
+                    LOG.debug("Skipping record id: "
+                        f"{redcap_record_instance.get('subject_id')}, "
+                        " encounter: "
+                        f"{redcap_record_instance.get('event_name')}, "
+                        ": insufficient information to construct encounter")
+                    continue
+
+        # We should never have to consult this
+        location_site_map = {
+            'bothell':  'UWBothell',
+            'odegaard': 'UWOdegaardLibrary',
+            'slu':      'UWSouthLakeUnion',
+            'tacoma':   'UWTacoma',
+            'uw_club':  'UWClub'
+            }
+
+        # TODO: since there's only one collection method
+        #  (courier-delivered swabs) I believe this will always be the same.
+        site_reference = create_site_reference(
+            location = None,
+            site_map = location_site_map,
+            default_site = SWAB_AND_SEND_SITE,
+            system_identifier = INTERNAL_SYSTEM)
+
+        # Handle various symptoms.
+        contained: List[dict] = []
+        diagnosis: List[dict] = []
+
+        # Map the various symptoms variables to their onset date.
+        # For daily_symptoms_covid_like we don't know the actual onset date. The questions asks
+        # "in the past 24 hours"
+        # TODO: verify with study
+        if event_type == EventType.ENCOUNTER:
+            symptom_onset_map = {
+                # sic--this misspelling is in the redcap form.
+                'symptoms': redcap_record_instance['ss_ealiest_date'],
+            }
+        # Irrelevant, no symptom questions are present in enrollment
+        elif event_type == EventType.ENROLLMENT:
+            symptom_onset_map = {'symptoms_base': redcap_record_instance['wk_symp_start_date']}
+        # Should never get here due to the event_type check at the top of the loop
+        else:
+            LOG.error(f"Invalid event_type {event_type} fell through")
```

Comment:
Are ss_ealiest_date and wk_symp_start_date both from the repeating weekly encounters? I 
I the referenced think daily_symptoms_covid_like in the comment is a holdover from HCT.

Is the enrollment instrument current_symptoms applicable here?  I'm not sure if it collects a start date though. If there isn't one, we could potentially use the date of that instrument as a best effort, but that might cause other data consistency issues 🤔

Reply:
Yes, I removed the comment referring to `daily_symptoms_covid_like`.